### PR TITLE
REF: Avoid dispatching Series ops to pd.Index

### PR DESCRIPTION
**Pull request by jbrockmendel**
We have some semi-circular dispatch logic between non-EA Series and pd.Index ops.  This avoids that circularity by implementing the relevant logic directly in the Series implementation.  After this I'm going to start migrating the Index implementations to use the Series implementation, after which we will have One True Implementation (also have to migrate RangeIndex and IntegerArray implementations)

### CODE REVIEW

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -1716,10 +1717,32 @@ def wrapper(left, right):
...
             #  that may incorrectly raise TypeError when we
             #  should get NullFrequencyError
-            result = op(pd.Index(left), right)
-            return construct_result(
-                left, result, index=left.index, name=res_name, dtype=result.dtype
-            )
+            orig_right = right
+            if is_scalar(right):
+                # broadcast and wrap in a TimedeltaIndex
+                # TODO: possible perf improvement by using NaTD scalar
```
Comment (jreback) [inline pandas/core/ops/__init__.py]: I don't think this this is a worthwhile comment here as NaTD is not likely to happen

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, otherwise lgtm.

File: pandas/core/ops/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: I think there's a decent chance we'll implement it for internal use to de-duplicate logic like this.

Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: Removed comment

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**